subgraphs/_reference_/src/sdk/protocols/bridge/pool.ts

```diff
@@ -0,0 +1,638 @@
+    route: PoolRoute,
+    eventType: TransactionType
+  ): void {
+    this.addVolume(
+      transfer.isOutgoing,
+      route,
+      transfer.amount,
+      transfer.amountUSD
+    );
+    this.protocol.addTransaction(eventType);
+
+    if (this.pool.type == BridgePoolType.BURN_MINT) {
+      let amount = transfer.amount;
+      if (transfer.isOutgoing) {
+        amount = amount.times(BIGINT_MINUS_ONE);
+      }
+      this.addMintSupply(amount);
+    } else if (
+      this.pool.type == BridgePoolType.LIQUIDITY ||
+      this.pool.type == BridgePoolType.LOCK_RELEASE
+    ) {
+      let amount = transfer.amount;
+      if (!transfer.isOutgoing) {
+        amount = amount.times(BIGINT_MINUS_ONE);
+      }
+      this.addInputTokenBalance(amount);
+    }
+  }
+
+  /**
+   * Will update the pool inputTokenBalance and TVL (also protocol TVL), based on the amount deposited.
+   * This will be called automatically if the LiquidityDeposit is created via an `Account`.
+   * It WON'T update the outputTokenSupply, so you should do that manually.
+   * @param deposit
+   * @see Account
+   */
+  trackDeposit(deposit: LiquidityDeposit): void {
+    this.addInputTokenBalance(deposit.amount);
+  }
+
+  /**
+   * Will update the pool inputTokenBalance and TVL (also protocol TVL), based on the amount withdrawn.
+   * This will be called automatically if the LiquidityWithdraw is created via an `Account`.
+   * It WON'T update the outputTokenSupply, so you should do that manually.
+   * @param deposit
+   * @see Account
+   */
+  trackWithdraw(withdraw: LiquidityWithdraw): void {
+    this.addInputTokenBalance(withdraw.amount.times(BIGINT_MINUS_ONE));
+  }
+}
+
+/**
+ * Will determine what's the type of the pool on the other side of the route.
+ * Possible combinations are:
+ * - BURN_MINT <> LOCK_RELEASE
+ * - BURN_MINT <> BURN_MINT
+ * - LIQUIDITY <> LIQUIDITY
+ *
+ * @param poolType the type of the pool on the current side of the route
+ * @param token the token sent through this route
+ * @returns {BridgePoolType}
+ */
+function inferCounterType(
+  poolType: BridgePoolType,
+  token: CrosschainToken
+): BridgePoolType {
+  if (poolType == BridgePoolType.LIQUIDITY) {
+    return BridgePoolType.LIQUIDITY;
+  }
+  if (poolType == BridgePoolType.LOCK_RELEASE) {
+    return BridgePoolType.BURN_MINT;
+  }
+  if (poolType == BridgePoolType.BURN_MINT) {
+    return token.type == CrosschainTokenType.WRAPPED
+      ? BridgePoolType.BURN_MINT
+      : BridgePoolType.LOCK_RELEASE;
+  }
+
+  log.error("Unknown pool type at inferCounterType {}", [poolType]);
+  log.critical("", []);
+  return poolType;
+}
+
+// A function which given 3 arrays of arbitrary types of the same length,
+// where the first one holds the reference order, the second one holds the same elements
+// as the first but in different order, and the third any arbitrary elements. It will return
+// the third array after sorting it according to the order of the first one.
```

Comment:
It'd helpful to add example inputs and output:
`sortArrayByReference(['a', 'c', 'b'], ['a', 'b', 'c'], [1, 2, 3]) => [1, 3, 2]`

Personally, I prefer switch the first and second argument.